### ISSUE TITLE
won't run

### ISSUE THREAD

**Issue by rogueturnip** [aeirola/react-native-svg-app-icon#13]
Hi!  Thanks for working on this project!  very useful.

I was trying to make it run and I've put an icon.svg in the root and run the command as indicated in the readme and go this error:
$ npx react-native-svg-app-icon
npx: installed 1 in 1.788s
Path must be a string. Received undefined
C:\Users\tony\desktop\code\project\node_modules\react-native-svg-app-icon\lib\cli.js
Unexpected reserved word

Thanks!

**Comment by rogueturnip** [comment]
I also tried it with ubuntu wsl and got
+ npx@10.2.0
added 484 packages from 651 contributors in 25.537s
tony@Tony-surface-pro ~/code/project(dev) $ npx react-native-svg-app-icon
/mnt/c/users/tony/desktop/code/project/node_modules/react-native-svg-app-icon/lib/cli.js:19
  for await (let file of generatedFiles) {
      ^^^^^

SyntaxError: Unexpected reserved word
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:617:28)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
    at bootstrap_node.js:625:3

**Comment by aeirola** [comment]
Hi, thanks for the interest in the tool. Which version of node are you running? Only versions 10 and later are supported at the moment. I realize that this could be better documented in both the readme and error message.

**Comment by rogueturnip** [comment]
Thanks.  I'm on 8 right now.  I will test with 10.

**Comment by aeirola** [comment]
Were you able to get it working on node 10? I'll update the readme to indicate the supported versions of node.

**Comment by rogueturnip** [comment]
That was definitely the problem.  However now I get hung up on not finding an app.json (I'm not running expo)
Running react-native-svg-app-icon
Reading file ./icon.svg
Wrote android\app\src\main\res\mipmap-mdpi\ic_launcher_round.png
Wrote android\app\src\main\res\mipmap-hdpi\ic_launcher_round.png
Wrote android\app\src\main\res\mipmap-xhdpi\ic_launcher_round.png
Wrote android\app\src\main\res\mipmap-xxhdpi\ic_launcher_round.png
Wrote android\app\src\main\res\mipmap-xxxhdpi\ic_launcher_round.png
Wrote android\app\src\main\res\drawable-anydpi-v26\ic_launcher_foreground.xml
Wrote android\app\src\main\res\mipmap-anydpi-v26\ic_launcher.xml
Wrote android\app\src\main\res\mipmap-anydpi-v26\ic_launcher_round.xml
(node:9632) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open 'C:\Users\tony\Desktop\code\alana-mobile\app.json'
(node:9632) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:9632) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
fs.js:114

**Comment by aeirola** [comment]
I'm sorry that you are having so many issues with the tool, I'll try to make it more straight-forward to use in the future. I've assumed that `app.json` is a part of standard react native installations, as it is [present](https://github.com/react-native-community/rn-diff-purge/blob/version/0.59.9/RnDiffApp/app.json) in newly created react native applications. What version of react native are you running? What has been the migration path for your application?

The tool uses `app.json` to identify the correct directory to add the iOS icon assets into. I guess it would be possible to do without the `app.json` file as well. Now that it seems like all react native projects do not necessarily include the file, that would probably be the right thing to do. I should probably try to mimic the behaviour of the react native cli tooling for identifying the ios project directory.

In the meantime, you can create an `app.json` file with the following content:

```json
{
  "name": "AppName"
}
```

where `AppName` is the name of the directory under the `ios` folder containing the `Images.xcassets` asset collection.

**Comment by rogueturnip** [comment]
We use to have an app.json when we worked with expo.  Since ejecting there wasn't any reason to keep it so we removed it.  Right now we are running on RN 0.59.9.

I'll try creating an app.json and see if that works for me.  I had assume it would detect the folders the same way app-icon did.

Thanks for creating this, I have no problem with this issues if I can help you move it forward to something the rest of the community can use.
